Tests/Adyen Tests/Analytics/AnalyticsProviderTests.swift

```diff
@@ -163,6 +192,221 @@ class AnalyticsProviderTests: XCTestCase {
+        sut?.sendInitialAnalytics(with: .components(type: .achDirectDebit), additionalFields: nil)
+        wait(for: .milliseconds(100))
+        
+        for _ in 0...2 {
+            sut?.add(info: AnalyticsEventInfo(component: "card", type: .rendered))
+        }
+        
+        let shouldSendExpectation = expectation(description: "send event is called")
+        apiClient.onExecute = { request in
+            if let analyticsRequest = request as? AnalyticsRequest {
+                XCTAssertEqual(analyticsRequest.infos.count, 3)
+                shouldSendExpectation.fulfill()
+            }
+        }
+        
+        sut = nil
+        
+        wait(for: [shouldSendExpectation], timeout: 1)
+    }
+    
+    func testAddingInfoShouldNotTriggerSend() {
+        let apiClient = APIClientMock()
+        let sut = AnalyticsProvider(apiClient: apiClient,
+                                    configuration: AnalyticsConfiguration(),
+                                    eventDataSource: eventDataSource)
+
+        let initialAnalyticsResponse = InitialAnalyticsResponse(checkoutAttemptId: checkoutAttemptIdMockValue)
+        let checkoutAttemptIdResult: Result<Response, Error> = .success(initialAnalyticsResponse)
+        
+        let analyticsResponse = AnalyticsResponse()
+        let analyticsResult: Result<Response, Error> = .success(analyticsResponse)
+        
+        apiClient.mockedResults = [checkoutAttemptIdResult, analyticsResult]
+        
+        sut.sendInitialAnalytics(with: .components(type: .achDirectDebit), additionalFields: nil)
+        wait(for: .milliseconds(100))
+        
+        let networkRequestExpectation = expectation(description: "send event should not be called")
+        networkRequestExpectation.isInverted = true
+        apiClient.onExecute = { request in
+            networkRequestExpectation.fulfill()
+        }
+        
+        let infoEvent = AnalyticsEventInfo(component: "card", type: .rendered)
+        sut.add(info: infoEvent)
+        
+        wait(for: [networkRequestExpectation], timeout: 1)
+    }
+    
+    func testAddingLogEventShouldTriggerSend() {
+        let apiClient = APIClientMock()
+        let sut = AnalyticsProvider(apiClient: apiClient,
+                                    configuration: AnalyticsConfiguration(),
+                                    eventDataSource: eventDataSource)
+
+        let initialAnalyticsResponse = InitialAnalyticsResponse(checkoutAttemptId: checkoutAttemptIdMockValue)
+        let checkoutAttemptIdResult: Result<Response, Error> = .success(initialAnalyticsResponse)
+        
+        let analyticsResponse = AnalyticsResponse()
+        let analyticsResult: Result<Response, Error> = .success(analyticsResponse)
+        
+        apiClient.mockedResults = [checkoutAttemptIdResult, analyticsResult]
+        
+        sut.sendInitialAnalytics(with: .components(type: .achDirectDebit), additionalFields: nil)
+        wait(for: .milliseconds(100))
```

Comment:
is there any way around waiting for a "random" amount of time?

Reply:
not sure really, maybe only testing through apiClient.onExecute? we can check next